v2/pkg/protocols/headless/engine/page.go

```diff
@@ -14,6 +15,13 @@ type Page struct {
 	rules    []requestRule
 	instance *Instance
 	router   *rod.HijackRouter
+	History  []HistoryData
```

Comment:
This history slice if being used concurrently will not be thread-safe unless we use a mutex. Do you think it's a good idea to add a mutex here @Mzack9999?

Reply:
done